File: GFG/Array/Day3/UnionOf2SortedArray.java
import java.util.ArrayList;

public class UnionOf2SortedArray {
    public static ArrayList<Integer> findUnion(int a[], int b[]) {
        
        ArrayList<Integer> res = new ArrayList<>();
        
        int i = 0, j = 0;

        while (i < a.length && j < b.length) {

            
            while (i > 0 && i < a.length && a[i] == a[i - 1]) {
                i++;
            }

            
            while (j > 0 && j < b.length && b[j] == b[j - 1]) {
                j++;
            }

            
            if (i >= a.length || j >= b.length) {
                break;
            }

            if (a[i] < b[j]) {
                res.add(a[i]);
                i++;
            }
            else if (a[i] > b[j]) {
                res.add(b[j]);
                j++;
            }
            else {
                res.add(a[i]);
                i++;
                j++;
            }
        }

        
        while (i < a.length) {
            if (i == 0 || a[i] != a[i - 1]) {
                res.add(a[i]);
            }
            i++;
        }

        
        while (j < b.length) {
            if (j == 0 || b[j] != b[j - 1]) {
                res.add(b[j]);
            }
            j++;
        }

        return res;
    }
}

File: GFG/Array/Day3/UnionOf2SortedArray.py
class Solution:
    def findUnion(self, a, b):
        # code here 
        i=0
        j=0
        res=[]
        
        while(i<len(a) and j<len(b)):
            if(i>0 and a[i]==a[i-1]):
                i+=1
                continue
            
            if(j>0 and b[j]==b[j-1]):
                j+=1
                continue
            
            if(i>len(a) or j>len(b)):
                break
            
            if(a[i]<b[j]):
                res.append(a[i])
                i+=1
            
            elif(b[j]<a[i]):
                res.append(b[j])
                j+=1
            
            else:
                res.append(a[i])
                i+=1
                j+=1
                
        while(i<len(a)):
            if(i>0 and a[i]==a[i-1]):
                i+=1
                continue
            res.append(a[i])
            i+=1
            
        while(j<len(b)):
            if(j>0 and b[j]==b[j-1]):
                j+=1
                continue
            res.append(b[j])
            j+=1
            
        return res